group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroupMember.java

```diff
@@ -571,15 +577,29 @@ public ConsumerGroupDescribeResponseData.Member asConsumerGroupDescribeMember(As
     }
 
     private static List<ConsumerGroupDescribeResponseData.TopicPartitions> topicPartitionsFromMap(
-        Map<Uuid, Set<Integer>> partitions
+        Map<Uuid, Set<Integer>> partitions,
+        Map<String, TopicMetadata> subscriptionMetadata
     ) {
         return partitions.entrySet().stream().map(
             item -> new ConsumerGroupDescribeResponseData.TopicPartitions()
                 .setTopicId(item.getKey())
+                .setTopicName(lookupTopicNameById(item.getKey(), subscriptionMetadata))
                 .setPartitions(new ArrayList<>(item.getValue()))
         ).collect(Collectors.toList());
     }
 
+    private static String lookupTopicNameById(
+        Uuid topicId,
+        Map<String, TopicMetadata> subscriptionMetadata
+    ) {
+        for (TopicMetadata topicMetadata : subscriptionMetadata.values()) {
+            if (topicId.equals(topicMetadata.id())) {
+                return topicMetadata.name();
+            }
+        }
+        throw UNKNOWN_TOPIC_ID.exception("The subscription metadata does not contain " + topicId);
```

Comment:
I am not sure about this. It means that when a topic is delete and the group has not been updated yet, it won't be possible to describe it. An alternative would be to directly remove the topic from the assignment. It will be eventually removed anyway. What do you think?